src/ppl/src/IOPlacer.cpp

```diff
@@ -758,6 +758,14 @@ bool IOPlacer::assignPinsToSections(int assigned_pins_count)
   if (total_pins_assigned == net->numIOPins()) {
     logger_->info(PPL, 8, "Successfully assigned pins to sections.");
     return true;
+  } else if (total_pins_assigned > net->numIOPins()) {
+    logger_->error(
+        PPL,
+        13,
+        "Internal error, placed more pins than exist ({} out of {}).",
+        total_pins_assigned,
+        net->numIOPins());
```

Comment:
I would suggest moving this to it's own if block above the first check 758. That way we conceptually guarantee that code after that point "isn't off the rails"